client/my-sites/plans-features-main/index.tsx

```diff
@@ -302,8 +305,24 @@ const PlansFeaturesMain = ( {
 			}
 		}
 
+		const selectedStorageOption =
+			selectedStorageOptions && selectedStorageOptions[ cartItemForPlan?.product_slug || '' ];
+		const storageAddOn = storageAddOns?.find( ( addOn ) => {
+			return addOn?.featureSlugs?.includes( selectedStorageOption || '' );
+		} );
+
+		const storageAddOnItemForPlan = storageAddOn && {
+			product_slug: storageAddOn.productSlug,
+			quantity: storageAddOn.quantity,
+			volume: 1,
+		};
+
 		if ( onUpgradeClick ) {
-			onUpgradeClick( cartItemForPlan );
+			const cartItems = [
+				...( cartItemForPlan ? [ cartItemForPlan ] : [] ),
+				...( storageAddOnItemForPlan ? [ storageAddOnItemForPlan ] : [] ),
+			];
```

Comment:
This logic feels a little disconnected to me. I think either none of it should happen here and instead go into the `handleUpgradeClick` of the grid components, OR unify everything (plan & add-on selection) to go through the
store.

I'd have expected these to be unified:

1. user interacts with grid and store gets updated with selections
2. on plan upgrade, the elements are put together sent through the callback here for processing OR both selected plan and add-ons are accessed through the data storehere

VS

1. user interacts with grid and store gets updated with the selections
2. on plan upgrade, plan is sent through as argument and wrapper queries the store for the
selected ado-ons

The side effect is `plans-features-main` accessing the plans-ui data store. I'd rather if we either remove the side-
effect or unify everything behind it.

Reply:
obviously opinion. might be a different angle to it

Reply:
> This logic feels a little disconnected to me. I think either none of it should happen here and instead go into the handleUpgradeClick of the grid components, OR unify everything (plan & add-on selection) to go through the store.

Hi @chriskmnds! Thanks for bringing this up -- to be honest, I spent so much time wrapping my head around signup plans code yesterday and fixing bugs that I forgot about your comment here https://github.com/Automattic/wp-calypso/pull/81340#discussion_r1329965409. It's my mistake. I'm thinking about your comment now 🤔

Reply:
> The side effect is plans-features-main accessing the plans-ui data store. I'd rather if we either remove the side-effect or unify everything behind it.

> I think either none of it should happen here and instead go into the handleUpgradeClick of the grid components, OR unify everything (plan & add-on selection) to go through the store.

> on plan upgrade, the elements are put together sent through the callback here for processing OR both selected plan and add-ons are accessed through the data storehere

Hey @chriskmnds  -- I think I might not fully understand the vision here, which is making it difficult for me to understand what we're intending to solve. Here's what I understand about what you're trying to convey:

- _Either_ move logic to build `cartItems` from the `handleUpgradeClick` in `plans-features-main` into the `handleUpgradeClick` in `plan-features-2023-grid`, or handle _both_ plan and add-on selection in the `wpcom-ui-store`.
- Using the data store for both plan and add-on selection makes sense to me, in that it means there wouldn't be two forms of state management being consolidated into `handleUpgradeClick` in `plans-features-main`
- The confusion for me is in If we migrate logic into `handleUpgradeClick` in `plan-features-2023-grid`. Are you also saying that, once we've moved the logic into `plan-features-2023-grid`, we'd want to use local state management to handle add-ons selection alongside plan selection? This is something that seems like it'd be a step backwards if that's the case. 🤔